code/modules/projectiles/guns/specialist.dm

```diff
@@ -715,121 +694,47 @@
 	icon_state = "m81"
 	item_state = "m81"
 	max_shells = 1 //codex
-	caliber = CALIBER_40MM //codex
-	load_method = SINGLE_CASING //codex
-	w_class = WEIGHT_CLASS_BULKY
 	flags_equip_slot = ITEM_SLOT_BACK|ITEM_SLOT_BELT
-	throw_speed = 2
-	throw_range = 10
-	force = 5.0
 	wield_delay = 0.2 SECONDS
-	fire_sound = 'sound/weapons/guns/fire/grenadelauncher.ogg'
-	fire_rattle = 'sound/weapons/guns/fire/grenadelauncher.ogg'
-	cocked_sound = 'sound/weapons/guns/interact/m92_cocked.ogg'
 	aim_slowdown = 1
-	gun_skill_category = GUN_SKILL_FIREARMS
 	flags_gun_features = GUN_UNUSUAL_DESIGN|GUN_AMMO_COUNTER
 	attachable_allowed = list()
-	general_codex_key = "explosive weapons"
 	attachable_offset = list("muzzle_x" = 33, "muzzle_y" = 18,"rail_x" = 14, "rail_y" = 22, "under_x" = 19, "under_y" = 14, "stock_x" = 19, "stock_y" = 14)
 	fire_delay = 1.05 SECONDS
-	/// The loaded grenade
-	var/obj/item/explosive/grenade/grenade
-	/// What type of grenade can be loaded
+	max_grenades = 1
 	var/grenade_type_allowed = /obj/item/explosive/grenade
```

Comment:
like i said above this should probably be a list of allowed grenades at the top level of GLs